More beautiful linear_gradient sample

goderbauer opened the pull request:
Taking over PR https://github.com/flutter/flutter/pull/96491 and enabling golden tests for API doc sample tests to make the newly added test pass.

Commit:
Update linear_gradient.0.dart

Tweaking example so that it's more attractive to a design-orientated audience

skia-gold commented:
Gold has detected about 1 new digest(s) on patchset 1.
View them at https://flutter-gold.skia.org/cl/github/99298

goderbauer commented:
@Piinks @gspencergoog This is enabling goldens testing for API samples for https://github.com/flutter/flutter/pull/96491. Can you take a look if that's the correct way to do it?

Piinks commented on examples/api/test/painting/linear_gradient.0_test.dart:
```diff
@@ -0,0 +1,39 @@
+    );
+    await expectLater(
+      find.byType(example.MoodyGradient),
+      matchesGoldenFile('linear_gradient.0_test.png'),
```
Can you include `api` in the name so it can be distinguished on the Gold dashboard? 
Maybe

```suggestion
      matchesGoldenFile('api.linear_gradient.0_test.png'),
```

goderbauer replied:
Ideally, we would configure this prefix globally in something like `flutter_test_config.dart`, so not everybody writing a test has to remember to put this here. Do you think that would be doable?

Piinks replied:
I think so... I am guessing it would be best to add in the same place we added an assertion in https://github.com/flutter/flutter/pull/99082
If the path contains `api`, `_addPrefix` can include `api` in the prefix. Do you want to add that here, or I can in a separate change.

goderbauer replied:
Done.

Piinks commented:
Looks like it works! 🎉 
![image](https://user-images.githubusercontent.com/16964204/156070220-83666b84-0646-463d-beda-bdca013fcb02.png)

goderbauer committed:
add api prefix

skia-gold commented:
Gold has detected about 1 new digest(s) on patchset 2.
View them at https://flutter-gold.skia.org/cl/github/99298

goderbauer committed:
add test

goderbauer commented:
@timsneath I incorporated your PR https://github.com/flutter/flutter/pull/96491 into this one and added the necessary configuration to make the golden test that you added pass. Can you LGTM this PR if you're OK with that? Otherwise, feel free to incorporate these changes into your original PR and we can submit that one.

skia-gold commented:
Gold has detected about 2 new digest(s) on patchset 3.
View them at https://flutter-gold.skia.org/cl/github/99298

goderbauer commented:
@timsneath said via chat that he is okay with this moving forward in this PR.

@Piinks wanna review this?

Piinks commented on packages/flutter_goldens/lib/flutter_goldens.dart:
```diff
@@ -25,15 +25,18 @@ const String _kFlutterRootKey = 'FLUTTER_ROOT';
   const Platform platform = LocalPlatform();
   if (FlutterPostSubmitFileComparator.isAvailableForEnvironment(platform)) {
-    goldenFileComparator = await FlutterPostSubmitFileComparator.fromDefaultComparator(platform);
+    goldenFileComparator = await FlutterPostSubmitFileComparator.fromDefaultComparator(platform, namePrefix: namePrefix);
```
Oh nice. It will probably scale better to other packages to have them set a prefix in the test config file. 👍

Piinks reviewed (approved):
LGTM! Let there be goldens! :)

Piinks commented:
The Google failure is unrelated to this change, manually set green.
